pkg/email/mailpit/client.go

```diff
@@ -0,0 +1,115 @@
+package mailpit
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	urlPkg "net/url"
+
+	"github.com/justtrackio/gosoline/pkg/appctx"
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+//go:generate mockery --name Client
+type Client interface {
+	ListMessages(ctx context.Context) (*ListMessagesResponse, error)
+	GetMessage(ctx context.Context, id string) (*Message, error)
+}
+
+type client struct {
+	client http.Client
+	config Config
+}
+
+type Config struct {
+	Server   string `cfg:"server"`
+	Protocol string `cfg:"protocol" default:"http"`
+}
+
+type mailpitCtxKey string
+
+func ProvideClient(ctx context.Context, config cfg.Config, _ log.Logger) (Client, error) {
+	return appctx.Provide(ctx, mailpitCtxKey("default"), func() (Client, error) {
+		var conf Config
+		config.UnmarshalKey("mailpit", &conf)
+
+		if conf.Server == "" {
+			return nil, fmt.Errorf("mailpit server is required")
+		}
+
+		return NewClientWithInterfaces(http.Client{}, conf), nil
+	})
+}
+
+func NewClientWithInterfaces(httpClient http.Client, conf Config) *client {
+	return &client{
+		client: httpClient,
+		config: conf,
+	}
+}
+
+func (c client) ListMessages(_ context.Context) (*ListMessagesResponse, error) {
```

Comment:
Can you put the mailpit for consistency into the test env for now? And please check the github actions ;)